sway-lib-std/src/bytes.sw

```diff
@@ -0,0 +1,849 @@
+        self.len
+    }
+
+    /// Clears the Bytes, removing all values.
+    ///
+    /// Note that this method has no effect on the allocated capacity
+    /// of the Bytes.
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// use std:bytes::Bytes;
+    ///
+    /// let bytes = Bytes::new();
+    /// bytes.push(5);
+    /// bytes.clear()
+    /// assert(bytes.is_empty());
+    /// ```
+    pub fn clear(ref mut self) {
+        self.len = 0;
+    }
+
+    /// Returns `true` if the vector contains no elements.
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// use std:bytes::Bytes;
+    ///
+    /// let bytes = Bytes::new();
+    /// assert(bytes.is_empty());
+    /// bytes.push(5);
+    /// assert(!bytes.is_empty());
+    /// bytes.clear()
+    /// assert(bytes.is_empty());
+    /// ```
+    pub fn is_empty(self) -> bool {
+        self.len == 0
+    }
+}
+
+// Need to use seperate impl blocks for now: https://github.com/FuelLabs/sway/issues/1548
+impl Bytes {
+    /// Creates a Bytes from a Vec<u8>.
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// use std:bytes::Bytes;
+    ///
+    /// let mut vec = Vec::new();
+    /// let a = 5u8;
+    /// let b = 7u8;
+    /// let c = 9u8
+    ///
+    /// vec.push(a);
+    /// vec.push(b);
+    /// vec.push(c);
+    ///
+    /// let bytes = Bytes::from_vec_u8(vec);
+    ///
+    /// assert(bytes.len == 3);
+    /// assert(bytes.get(0).unwrap() == a);
+    /// assert(bytes.get(1).unwrap() == b);
+    /// assert(bytes.get(2).unwrap() == c);
+    /// ```
+    pub fn from_vec_u8(ref mut vec: Vec<u8>) -> Self {
+        let mut bytes = Bytes::new();
+        let mut i = 0;
+        let length = vec.len();
+        while i < length {
+            bytes.push(vec.get(i).unwrap());
+            i += 1;
+        };
+        bytes
+    }
+
+    /// Creates a Vec<u8> from a Bytes.
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// use std:bytes::Bytes;
+    ///
+    /// let mut bytes = Bytes::new();
+    /// let a = 5u8;
+    /// let b = 7u8;
+    /// let c = 9u8
+    /// bytes.push(a);
+    /// bytes.push(b);
+    /// bytes.push(c);
+    ///
+    /// assert(bytes.len() == 3);
+    ///
+    /// let vec = bytes.into_vec_u8();
+    ///
+    /// assert(vec.len() == 3);
+    /// assert(vec.get(0).unwrap() == a);
+    /// assert(vec.get(1).unwrap() == b);
+    /// assert(vec.get(2).unwrap() == c);
+    /// ```
+    pub fn into_vec_u8(self) -> Vec<u8> {
+        let mut vec = Vec::new();
+        let mut i = 0;
+        let length = self.len;
+        while i < length {
+            vec.push(self.get(i).unwrap());
+            i += 1;
+        };
+        vec
+    }
+
+    pub fn split(self, index: u64) -> (Bytes, Bytes) {
+        // TODO: handle edge cases:
+        // index == 0 (also handles the self.len() == 1 case)
+        // index == self.len() - maybe we say index must be < self.len - 1
+        // @todo check rust impl for this type of thing
```

Comment:
Should these todos be tracked in an issue?

Reply:
Ah, I will review these and either remove or track them in issues as needed.